targets/TARGET_Infineon/tools/RingBuff.h

```diff
@@ -0,0 +1,179 @@
+/* mbed Example Program
+ * Copyright (c) 2006-2014 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// library provided by https://github.com/wizard97/Embedded_RingBuf_CPP
+
+#ifndef MBED_OS_CIRCULAR_BUFFER_H_
+#define MBED_OS_CIRCULAR_BUFFER_H_
+
+#include "mbed.h"
+#include "rtos.h"
```

Comment:
include what is needed , rtos and mbed header files are for an application space. This would cause inclusion of unnecessary files